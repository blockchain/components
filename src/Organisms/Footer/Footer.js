// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

import { screenSize } from '../../Tools/interpolation'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  width: 100%;
  padding: 1.25rem 2rem;
  box-sizing: border-box;

  @media (min-width: ${screenSize('sm')}) {
    padding: 1.25rem 3rem;
  }
`

type PropsType = {
  +children?: React.Node,
}

const Footer = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

Footer.propTypes = {
  children: PropTypes.node,
}

export default Footer
