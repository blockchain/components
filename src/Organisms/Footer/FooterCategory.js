import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.h5`
  color: ${(props) => props.theme['footerCategoryColor']};
  font-family: ${(props) => props.theme['fontPrimary']};
  font-size: 0.875rem;
  font-weight: 600;
  text-transform: uppercase;
  margin-bottom: 1rem;
`

const FooterCategory = ({ children, ...rest }) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

FooterCategory.propTypes = {
  children: PropTypes.node,
}

export default FooterCategory
