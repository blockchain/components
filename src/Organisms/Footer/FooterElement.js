import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  width: 100%;
  margin-top: 1.25rem;

  & > :first-child {
    margin-right: 0.75rem;
  }
`

const FooterElement = ({ children, ...rest }) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

FooterElement.propTypes = {
  children: PropTypes.node,
}

export default FooterElement
