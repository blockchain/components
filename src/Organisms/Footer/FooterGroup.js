// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

type PropsType = {
  +children?: React.Node,
}

const Wrapper = styled.div`
  flex: 0 1 auto;
  width: 8rem;
  margin-bottom: 4rem;
  margin-right: 2rem;

  @media (max-width: ${(props) => props.theme.screenSizes.sm}) {
    margin-bottom: 2rem;
  }
`

const FooterRow = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

FooterRow.propTypes = {
  children: PropTypes.node,
}

export default FooterRow
