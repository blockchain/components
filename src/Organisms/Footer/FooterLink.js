// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled, { type ReactComponentFunctional } from 'styled-components'

import Link, { type PropsType as LinkPropsType } from '../../Atoms/Links/Link'

export type PropsType = {
  +children?: React.Node,
}

const Wrapper: ReactComponentFunctional<LinkPropsType> = styled(Link).attrs({
  size: '1rem',
  weight: 500,
})`
  position: relative;
  white-space: nowrap;
`

const FooterLink = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

FooterLink.propTypes = {
  children: PropTypes.node.isRequired,
}

export default FooterLink
