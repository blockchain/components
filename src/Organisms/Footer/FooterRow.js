import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;

  @media (min-width: ${(props) => props.theme.screenSizes.sm}) {
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
    flex-wrap: wrap;
    width: 100%;
  }

  @media (min-width: ${(props) => props.theme.screenSizes.md}) {
    flex-wrap: no-wrap;
  }
`

const FooterRow = ({ children, ...rest }) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

FooterRow.propTypes = {
  children: PropTypes.node,
}

export default FooterRow
