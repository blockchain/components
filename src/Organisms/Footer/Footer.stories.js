import React from 'react'
import { storiesOf } from '@storybook/react'

import Footer from './Footer'
import FooterCategory from './FooterCategory'
import FooterElement from './FooterElement'
import FooterGroup from './FooterGroup'
import FooterRow from './FooterRow'

storiesOf('Organisms/Footer', module)
  .add('example 1', () => (
    <Footer>
      <FooterRow>
        <FooterGroup>
          <FooterCategory>Category 1</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
        <FooterGroup>
          <FooterCategory>Category 2</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
        <FooterGroup>
          <FooterCategory>Category 3</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
        <FooterGroup>
          <FooterCategory>Category 4</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
      </FooterRow>
    </Footer>
  ))
  .add('example 2', () => (
    <Footer>
      <FooterRow>
        <FooterGroup>
          <FooterCategory>Category 1</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
        <FooterGroup>
          <FooterCategory>Category 2</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
        <FooterGroup>
          <FooterCategory>Category 3</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
        <FooterGroup>
          <FooterCategory>Category 4</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
        <FooterGroup>
          <FooterCategory>Category 5</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
        <FooterGroup>
          <FooterCategory>Category 6</FooterCategory>
          <FooterElement>
            <a>Link1</a>
          </FooterElement>
          <FooterElement>
            <a>Link2</a>
          </FooterElement>
          <FooterElement>
            <a>Link3</a>
          </FooterElement>
        </FooterGroup>
      </FooterRow>
    </Footer>
  ))
