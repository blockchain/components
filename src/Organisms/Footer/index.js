// @flow strict
import Footer from './Footer'
import FooterCategory from './FooterCategory'
import FooterElement from './FooterElement'
import FooterGroup from './FooterGroup'
import FooterLink from './FooterLink'
import FooterRow from './FooterRow'

export {
  Footer,
  FooterCategory,
  FooterElement,
  FooterGroup,
  FooterLink,
  FooterRow,
}
