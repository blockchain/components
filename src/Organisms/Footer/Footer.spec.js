import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import Footer from './Footer'

describe('Footer', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Footer>Footer</Footer>)

    expect(tree.find(Footer)).toMatchSnapshot()
  })
})
