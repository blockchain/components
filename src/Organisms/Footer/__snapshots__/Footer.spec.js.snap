// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: 100%;
  padding: 1.25rem 2rem;
  box-sizing: border-box;
}

@media (min-width:48rem) {
  .c0 {
    padding: 1.25rem 3rem;
  }
}

<Footer>
  <styled.div>
    <div
      className="c0"
    >
      Footer
    </div>
  </styled.div>
</Footer>
`;
